TestProviders/AzureProvider.psm1

```diff
@@ -128,13 +137,27 @@ Class AzureProvider : TestProvider
 					Write-LogErr "$($restartJob.Name) failed with error: ${jobError}"
 					return $false
 				} else {
-					$tempJobs += $restartJob
-					$recheckAgain = $true
+					if ((Get-Date) -gt $Timeout ) {
+						Write-LogErr "$($restartJob.Name) timed out after $TimeoutMinutes minutes. Removing the job."
+						$null = Remove-Job -Id $restartJob.ID -Force -ErrorAction SilentlyContinue
+						$TimedOutResourceGroup = $restartJob.Name.Replace("Restart-",'').Split(':')[0]
+						$TimedOutRoleName = $restartJob.Name.Replace("Restart-",'').Split(':')[1]
+						$TimedOutVM = $AllVMData | Where-Object {$_.ResourceGroupName -eq $TimedOutResourceGroup -and $_.RoleName -eq $TimedOutRoleName}
+						$Null = Restart-VMFromShell -VMData $TimedOutVM -SkipRestartCheck
```

Comment:
why do we restart it from shell if Restart-AzureRmVM job timeout? can we retry Restart-AzureRmVM?

Reply:
This workflow already tried but didn't work.

Reply:
explain in detail about 'tried but did not work' please.

Reply:
@juhlee-microsoft, 
We attempted two remedies earlier, 
1. Increased the restart retries with larger interval 
https://github.com/LIS/LISAv2/commit/f6c731424d07db689ebcd6ff571a7ea06108dd73 

2. Replaced the Restart-AzureRmVM script block with in-built background restart operation.
https://github.com/LIS/LISAv2/commit/a5b1dcd0142e5d1df8fbb5c46d5c0402d7ab6c02

**Both attempts, didn't solve the problem.**

Reply:
@lizzha And another problem with timeout is - once the Restart-AzureRmVM operations times out, I can not see "Restart" button active on portal also. In that case, the only option is Stop VM -> Start VM. But if we stop the VM, then we might loose the PublicIP of the VM which will create further problems.